main.py

```diff
@@ -190,76 +189,57 @@ def extend_mw(self, request: Request) -> JSONResponse:
-            raise HTTPException(
-                400,
-                detail='Minutes of extension must be sent'
-            )
+            raise HTTPException(404, detail=f"Maintenance with id {mw_id} not found")
+        if "minutes" not in data:
+            raise HTTPException(400, detail="Minutes of extension must be sent")
         if maintenance.status == Status.PENDING:
             raise HTTPException(
-                400,
-                detail=f'Maintenance window {mw_id} has not yet started'
+                400, detail=f"Maintenance window {mw_id} has not yet started"
             )
         if maintenance.status == Status.FINISHED:
             raise HTTPException(
-                400,
-                detail=f'Maintenance window {mw_id} has already finished'
+                400, detail=f"Maintenance window {mw_id} has already finished"
             )
         try:
-            maintenance_end = maintenance.end + \
-                timedelta(minutes=data['minutes'])
-            new_maintenance = maintenance.copy(
-                update={'end': maintenance_end}
-            )
+            maintenance_end = maintenance.end + timedelta(minutes=data["minutes"])
+            new_maintenance = maintenance.copy(update={"end": maintenance_end})
         except TypeError as exc:
             raise HTTPException(
-                400,
-                detail='Minutes of extension must be integer') from exc
+                400, detail="Minutes of extension must be integer"
+            ) from exc
 
         self.scheduler.update(new_maintenance)
-        return JSONResponse({'response': f'Maintenance {mw_id} extended'})
+        return JSONResponse({"response": f"Maintenance {mw_id} extended"})
 
     def validate_item_existence(self, window: MW):
         """Validate that all items in a maintenance window exist."""
         non_existant_switches = list(
             filter(
-                lambda switch_id:
-                    self.controller.switches.get(switch_id)
-                    is None,
-                window.switches
+                lambda switch_id: self.controller.switches.get(switch_id) is None,
+                window.switches,
             )
         )
         non_existant_interfaces = list(
             filter(
-                lambda interface_id:
-                    self.controller.get_interface_by_id(interface_id)
-                    is None,
-                window.interfaces
+                lambda interface_id: self.controller.get_interface_by_id(interface_id)
+                is None,
+                window.interfaces,
             )
         )
         non_existant_links = list(
             filter(
-                lambda interface_id:
-                    self.controller.napps[('kytos', 'topology')]
-                    .links.get(interface_id)
-                    is None,
-                window.interfaces
+                lambda link_id: self.controller.napps[("kytos", "topology")].links.get(
```

Comment:
That's what I get for copy and pasting code. Thank you for catching it @HeriLFIU.